spyder/plugins/editor/widgets/status.py

```diff
@@ -30,6 +43,31 @@ def get_tooltip(self):
         """Return localized tool tip for widget."""
         return _("File permissions")
 
+    def toggle_rw_r(self):
+        """
+        Toggle the rw permission. Explicitily use the permission that
+        is displayed in the widget as basis for switch decision, to prevent
+        that the file permission has been changed in the background and not
+        been updated in Spyder (in which case it would be switched back again).
+        """
+        filename = self.parent.get_current_filename()
+        writeable = QFileInfo(filename).isWritable()
+        self.set_writable(not writeable)
+
+    def set_writable(self, writeable):
+        """set the current file to writeable or not writeable"""
+        filename = self.parent.get_current_filename()
+        try:
+            if writeable:
+                os.chmod(filename, S_IWUSR | S_IREAD)
+            else:
+                os.chmod(filename, S_IREAD | S_IRGRP | S_IROTH)
+            self.parent.editorstacks[0].refresh()
```

Comment:
I'm pretty sure my use of `self.parent.editorstacks[0]` is wrong, even though it will work in most cases. I found no other way to refresh all editorstacks or find out which is the current editor stack.

Reply:
There is a function `get_current_editorstack`.

Reply:
oh, strange that my autocomplete didn't find it, and I also could not locate it in `__dict__`. Thanks! added that